### ISSUE TITLE
Bug: New option to limit field depth is not working correctly

### ISSUE THREAD

**Issue by robross0606** [jakepusateri/graphql-list-fields#20]
The new feature request (#18) added on #19 does not appear to be working properly.  It appears to be ignoring fields "at target depth" that have further fields inside them, instead of just returning that field's name.  For example, if I do not set the new `depth` option by simply calling `getFieldNames(info)`, my AST returns this list of fields:
```
[
  "id",
  "_internalId",
  "revision",
  "__typename",
  "rawState",
  "effectiveState",
  "children.id",
  "children.subId",
  "children.rawState",
  "children.effectiveState",
  "children.studyActivitySubId",
  "children.children.id",
  "children.children.subId",
  "children.children.rawState",
  "children.children.effectiveState",
  "children.children.studyActivitySubId",
  "children.children.children.id",
  "children.children.children.subId",
  "children.children.children.rawState",
  "children.children.children.effectiveState",
  "children.children.children.studyActivitySubId",
  "children.children.nestedParent.id",
  "children.children.nestedParent._internalId",
  "children.children.nestedParent.revision",
  "children.children.nestedParent.__typename",
]
```
However, if I set depth to 1 with `getFieldNames(info, 1)`, the results are:
```
[
  "id",
  "_internalId",
  "revision",
  "__typename",
  "rawState",
  "effectiveState",
]
```

This is incorrect.  Note that `children` is not returned.  The correct results would be:
```
[
  "id",
  "_internalId",
  "revision",
  "__typename",
  "rawState",
  "effectiveState",
  "children",
]
```

It should not be returning anything _under_ `children` (e.g. `children.rawState`), but it should still return `children` in the list since it is a requested field at depth of 1.

**Comment by jakepusateri** [comment]
Yeah that’s a little tricky. “children” is not a field, so I’m not sure it
makes sense to include it. If we think of the query as a tree, we only ever
return leaf nodes aka fields. Returning an internal node aka selection set
is inconsistent.

I can imagine why you might want this, but if we return e.g. “children” the
caller has no way of knowing if it’s a real field or not.

It might make more sense to implement the filtering on your end after
calling getFieldNames with no depth limit.

**Comment by robross0606** [comment]
Just curious.  If you do not consider `children` to be a "field", then what is it?

**Comment by robross0606** [comment]
> It might make more sense to implement the filtering on your end after calling getFieldNames with no depth limit.

This is what I was already doing, so the requested feature as implemented is not very useful.

**Comment by jakepusateri** [comment]
I went and looked at the spec since it’s been a while.
https://spec.graphql.org/October2021/#sec-Language.Fields

Looks like you are correct and these are all fields. Some fields have
selection sets, but we should return them when listing fields. The caller
will just have to deal with that.

I won’t be able to get to this for a bit but could look at a pr if you
already have working code for this. The tests I added would need to be
modified since I was testing the wrong behavior.
